tests/e2e/specs/scaffold.spec.ts

```diff
@@ -1,94 +1,169 @@
+      const supportView = new SupportView(page);
+      // and we start the generate project flow
+      await supportView.clickTreeItem("Generate Project from Template");
+      // and we choose a project template from the quickpick
+      const projectQuickpick = new Quickpick(page);
+      await projectQuickpick.selectItemByText(templateDisplayName);
+      // and we provide a simple example configuration and submit the form
+      const scaffoldForm = new ProjectScaffoldWebview(page);
+      await scaffoldForm.bootstrapServersField.fill("localhost:9092");
+      await scaffoldForm.submitForm();
 
-        // Given we navigate to the Support view
-        const supportView = new SupportView(page);
-        // and we start the generate project flow
-        await supportView.clickTreeItem("Generate Project from Template");
-        // and we choose a project template from the quickpick
-        const projectQuickpick = new Quickpick(page);
-        await projectQuickpick.selectItemByText(templateName);
-        // and we provide a simple example configuration and submit the form
-        const scaffoldForm = new ProjectScaffoldWebview(page);
-        await scaffoldForm.bootstrapServersField.fill("localhost:9092");
+      // Then we should see that the project was generated successfully
+      await verifyGeneratedProject(page, templateName, "localhost:9092");
+    });
 
-        // When we submit the form
-        await scaffoldForm.submitForm();
-        // Then we should see that the project was generated successfully
-        const notificationArea = new NotificationArea(page);
-        const infoNotifications = notificationArea.infoNotifications.filter({
-          hasText: "Project Generated",
+    for (const [connectionType, connectionTag, connectionSetup] of connectionTypes) {
+      test.describe(`${connectionType} connection`, { tag: [connectionTag] }, () => {
+        test.beforeEach(async ({ page, electronApp }) => {
+          // set up the connection based on type
+          await connectionSetup(page, electronApp);
         });
-        await expect(infoNotifications).toHaveCount(1);
-        const successNotification = new Notification(page, infoNotifications.first());
-        await successNotification.clickActionButton("Open in Current Window");
 
-        // When we open the configuration file .env
-        const envFileName = ".env";
-        const explorerView = new View(page, "Explorer");
-        const envFile = explorerView.treeItems.filter({
-          hasText: envFileName,
+        test(`should apply ${templateDisplayName} template from Kafka topic in Topics view`, async ({
+          page,
+        }) => {
+          // Given we navigate to a topic in the Topics view
+          const topicsView = new TopicsView(page);
+          await topicsView.loadTopics(connectionType, SelectKafkaCluster.FromResourcesView);
+          await expect(topicsView.topics).not.toHaveCount(0);
```

Comment:
Not required for this branch, but we'll want to eventually migrate this to create+delete topics as well so we aren't relying on a persistent topic as part of https://github.com/confluentinc/vscode/issues/2602

Reply:
Makes a lot of sense. I've added a comment to that issue.